src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/languageConstructions/ProtectedWithFinalClassInspector.java

```diff
@@ -0,0 +1,139 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.languageConstructions;
+
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.navigation.NavigationItem;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.SmartPointerManager;
+import com.intellij.psi.SmartPsiElementPointer;
+import com.intellij.psi.impl.source.tree.LeafPsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.jetbrains.php.lang.psi.PhpPsiElementFactory;
+import com.jetbrains.php.lang.psi.elements.*;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpElementVisitor;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpInspection;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class ProtectedWithFinalClassInspector extends BasePhpInspection {
+    private static final String message = "Protected modifier could be replaced by private.";
+
+    @NotNull
+    public String getShortName() {
+        return "ProtectedWithFinalClassInspection";
+    }
+
+    @Override
+    @NotNull
+    public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder problemsHolder, final boolean isOnTheFly) {
+        return new BasePhpElementVisitor() {
+            private void checkElement(final PhpElementWithModifier element) {
+                if (!element.getModifier().isProtected()) {
+                    return;
+                }
+
+                final PhpClass elementClass = ((PhpClassMember) element).getContainingClass();
+
+                assert elementClass != null;
```

Comment:
Asserts are good option when we have sure that the value will never fail. For instance, in this assert point we have sure that the class exists, because it will be triggered only from field or method visitor (that are needly a class member). It avoid the NPE warning from IDE and the "return if null". The compiled version will drop the assert, then obvious checks will not be done at runtime. It's used a lot on the intellij API too. But, if you still prefer, I can replace the asserts.